src/app/services/coin.service.ts

```diff
@@ -0,0 +1,40 @@
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+import { BaseCoin } from '../coins/basecoin';
+import { SkycoinCoin } from '../coins/skycoin.coin';
+import { TestCoin } from '../coins/test.coin';
+
+export class CoinService {
+
+  currentCoin: BehaviorSubject<BaseCoin> = new BehaviorSubject<BaseCoin>(null);
+
+  private coins: BaseCoin[] = [];
+  private readonly storageKey = 'currentCoin';
+
+  constructor() {
+    this.loadCurrentCoin();
+    this.loadAvailableCoins();
+  }
+
+  changeCoin(coin: BaseCoin) {
+    this.currentCoin.next(coin);
+
+    this.saveCoin(coin);
+  }
+
+  private loadCurrentCoin() {
+    const storedCoin: string = localStorage.getItem(this.storageKey);
+    if (storedCoin) {
+      const coin: BaseCoin = JSON.parse(storedCoin);
+      this.currentCoin.next(coin);
+    }
```

Comment:
If it is not possible to obtain a coin from persistent storage, `this.currentCoin` should be set to the default coin.

Reply:
Fixed.